hamilton/experimental/h_spark.py

```diff
@@ -310,29 +301,95 @@ def _lambda_udf(
             node_.name, spark_udf(*[_value for _name, _value in params_from_df.items()])
         )
 
-    def _check_kwargs(
-        self, kwargs: Dict[str, Any]
-    ) -> Tuple[Union[DataFrame, GroupedData], Dict[str, Any]]:
-        """Sanitizes the kwargs to remove the dataframes"""
+    @staticmethod
+    def get_pandas_annotations(hamilton_udf: Callable) -> Dict[str, bool]:
+        """Given a function, return a dictionary of the parameters that are annotated as pandas series.
+
+        :param hamilton_udf: the function to check.
+        :return: dictionary of parameter names to boolean indicating if they are pandas series.
+        """
+        new_signature = inspect.signature(hamilton_udf)
+        new_sig_parameters = dict(new_signature.parameters)
+        pandas_annotation = {
+            name: param.annotation == pd.Series
+            for name, param in new_sig_parameters.items()
+            if param.default
+            == inspect.Parameter.empty  # bound parameters will have a default value.
+        }
+        return pandas_annotation
+
+    @staticmethod
+    def bind_parameters_to_callable(
```

Comment:
Wonder if we can employ [signature.bind](https://docs.python.org/3/library/inspect.html#inspect.Signature.bind) here...

Reply:
I don't think so -- that's only for the signature object, not the actual function?